ProfileService/ProfileService.BLL/Services/UserProfileService.cs

```diff
@@ -21,5 +21,29 @@ public async Task<UserProfile> CreateProfileAsync(UserRegistrationData userData,
 
             return await profileRepository.CreateAsync(userProfile, cancellationToken);
         }
+
+        public async Task<UserProfile> GetProfileAsync(Guid id, CancellationToken cancellationToken = default)
+        {
+            var userProfile = await profileRepository.GetByIdAsync(id, cancellationToken);
+
+            if (userProfile is null)
+            {
+                throw new NotFoundException(UserProfileMessages.ProfileNotFound);
+            }
```

Comment:
What if we will use .FirstAsync() instead of FirstOrDefaultAsync() in repository and handle exception thrown by it